docs/infrastructure/native-image.md

```diff
@@ -201,20 +201,71 @@ safely.
 ### Engine runner Configuration
 
 The Native Image generation for the Engine Runner is currently in a preview
-state. Limitations are currently mostly due to
-[Java interop](https://www.pivotaltracker.com/story/show/183260380) and loading
-of stdlib components. To generate the Native Image for runner simply execute
+state. To generate the Native Image for runner simply execute
 
-```
+```bash
 sbt> engine-runner/buildNativeImage
 ```
 
 and execute the binary on a sample factorial test program
 
-```
+```bash
 > runner --run engine/runner-native/src/test/resources/Factorial.enso 6
 ```
 
 The task that generates the Native Image, along with all the necessary
 configuration, reside in a separate project due to a bug in the currently used
-GraalVM version.
+GraalVM version. As September 2023 it can execute all Enso code, but cannot
+invoke `IO.println` or other library functions that require
+[polyglot java import](../../docs/polyglot/java.md), but read on...
+
+### Engine with Espresso
+
+Since [PR-6966](https://github.com/enso-org/enso/pull/6966) there is an
+experimental support for including
+[Espresso Java interpreter](https://www.graalvm.org/jdk17/reference-manual/java-on-truffle/)
+to allow use of some library functions (like `IO.println`) in the _Native Image_
+built runner.
+
+The support can be enabled by setting environment variable `ENSO_JAVA=espresso`
+and making sure Espresso is installed in GraalVM executing the Enso engine -
```

Comment:
```suggestion
and making sure Espresso is installed in GraalVM executing the build process -
```

Reply:
No, it is really GraalVM executing the engine - at least in JVM mode. The code is compiled in all the time and can be activated by setting `ENSO_JAVA=espresso` before running the Enso engine.